components/datetime/src/format/datetime.rs

```diff
@@ -403,164 +387,165 @@ where
-                    .map_err(Error::DateTimeInput)
-                    .map(|x| x.1 .0)
-                {
-                    Ok(x) => {
-                        r = r.and(try_write_number(
-                            w,
-                            fixed_decimal_format,
-                            FixedDecimal::from(x),
-                            field.length,
-                        )?)
-                    }
+                let () = match datetime.week_of_year().map_err(Error::DateTimeInput) {
+                    Ok(x) => r = r.and(try_write_number(w, fdf, x.1 .0.into(), field.length)?),
                     Err(e) => {
                         r = r.and(Err(e));
-                        w.with_part(Part::ERROR, |w| "<ICU4X ERROR>".write_to(w))?
+                        w.with_part(Part::ERROR, |w| "{week_of_year}".write_to(w))?
                     }
                 };
             }
             Week::WeekOfMonth => {
-                let () = match datetime
-                    .week_of_month()
-                    .map_err(Error::DateTimeInput)
-                    .map(|x| x.0)
-                    .map(FixedDecimal::from)
-                {
-                    Ok(x) => r = r.and(try_write_number(w, fixed_decimal_format, x, field.length)?),
+                let () = match datetime.week_of_month().map_err(Error::DateTimeInput) {
+                    Ok(x) => r = r.and(try_write_number(w, fdf, x.0.into(), field.length)?),
                     Err(e) => {
                         r = r.and(Err(e));
-                        w.with_part(Part::ERROR, |w| "<ICU4X ERROR>".write_to(w))?
+                        w.with_part(Part::ERROR, |w| "{week_of_month}".write_to(w))?
                     }
                 };
             }
         },
         FieldSymbol::Weekday(weekday) => {
-            let () = match date_symbols.ok_or(Error::MissingDateSymbols).and_then(|x| {
-                x.get_symbol_for_weekday(
-                    weekday,
-                    field.length,
-                    datetime
-                        .datetime()
-                        .iso_weekday()
-                        .ok_or(Error::MissingInputField(Some("iso_weekday")))?,
-                )
-            }) {
-                Ok(symbol) => w.write_str(symbol)?,
+            let wd = datetime
+                .datetime()
+                .iso_weekday()
+                .ok_or(Error::MissingInputField(Some("iso_weekday")));
+
+            let () = match date_symbols
+                .ok_or(Error::MissingDateSymbols)
+                .and_then(|x| x.get_symbol_for_weekday(weekday, field.length, wd?))
+            {
+                Ok(s) => w.write_str(s)?,
                 Err(e) => {
                     r = r.and(Err(e));
-                    w.with_part(Part::ERROR, |w| "<ICU4X ERROR>".write_to(w))?
+                    w.with_part(Part::ERROR, |w| {
+                        if let Ok(wd) = wd {
+                            "WD".write_to(w)?;
```

Comment:
Thought: maybe just `D`... then we end up with `M1` and `D1`. I think these are the only two cases where we print out a numeric fallback.

Reply:
I would interpret D1 M2 2024 as 2024-02-01, which is why I used `WD`.

Reply:
Maybe use root.xml data here, too. Hardcode the titlecase 3-letter names `"Sun"`, `"Mon"`, ...

https://github.com/unicode-org/cldr/blob/dd4a0bd49cc6dcc16ad7625d6ec7fcb583fb34d4/common/main/root.xml#L1201

Reply:
weird that the months have codes but the days don't